grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,465 @@
+                ==\
+                    ma_env._cent_env.shunt_to_subid[
+                        ma_env._subgrids_cls[space][agent].mask_shunt
+                    ]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_orig_ids[
+                        ma_env._subgrids_cls[space][agent].line_ex_to_subid]\
+                ==\
+                    ma_env._cent_env.line_ex_to_subid[
+                        ma_env._subgrids_cls[space][agent].mask_line_ex
+                    ]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_orig_ids[
+                        ma_env._subgrids_cls[space][agent].line_or_to_subid]\
+                ==\
+                    ma_env._cent_env.line_or_to_subid[
+                        ma_env._subgrids_cls[space][agent].mask_line_or
+                    ]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_orig_ids[
+                        ma_env._subgrids_cls[space][agent].interco_to_subid][
+                            ma_env._subgrids_cls[space][agent].interco_is_origin]\
+                ==\
+                    ma_env._cent_env.line_or_to_subid[
+                        ma_env._subgrids_cls[space][agent].mask_interco
+                    ][
+                        ma_env._subgrids_cls[space][agent].interco_is_origin
+                ]).all()
+            
+            assert (ma_env._subgrids_cls[space][agent].sub_orig_ids[
+                        ma_env._subgrids_cls[space][agent].interco_to_subid][
+                            ~ma_env._subgrids_cls[space][agent].interco_is_origin]\
+                ==\
+                    ma_env._cent_env.line_ex_to_subid[
+                        ma_env._subgrids_cls[space][agent].mask_interco
+                    ][
+                        ~ma_env._subgrids_cls[space][agent].interco_is_origin
+                ]).all()
+            
+            for subid in range(ma_env._subgrids_cls[space][agent].n_sub):
+                dict_local = ma_env._subgrids_cls[space][agent].get_obj_connect_to(substation_id=subid)
+                dict_global = self.env.get_obj_connect_to(
+                    substation_id=ma_env._subgrids_cls[space][agent].sub_orig_ids[subid]
+                )
+                
+                assert dict_local['nb_elements'] == dict_global['nb_elements']
+                assert (dict_global['loads_id']\
+                    ==\
+                        ma_env._subgrids_cls[space][agent].load_orig_ids[dict_local['loads_id']]
+                ).all()
+
+                assert (dict_global['generators_id']\
+                    ==\
+                        ma_env._subgrids_cls[space][agent].gen_orig_ids[dict_local['generators_id']]
+                ).all()
+                
+                assert (dict_global['storages_id']\
+                    ==\
+                        ma_env._subgrids_cls[space][agent].storage_orig_ids[dict_local['storages_id']]
```

Comment:
This part of the test should not start like this, it should read the full grid directly  and not read from sub_grid.n_sub

It should start `for sub_orig_id in range(env.n_sub): blablabla`

Reply:
Ok, done :)